test_scripts/Resumption/Handling_errors_from_HMI/050_resume_IVD_subscription_after_erroneous_response_all_moduleTypes.lua

```diff
@@ -0,0 +1,104 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0190-resumption-data-error-handling.md
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0188-get-interior-data-resumption.md
+--
+-- Description:
+-- In case:
+-- 1. Subscriptions for all interior vehicle data types are added
+-- 2. Unexpected disconnect and Reconnect are performed
+-- 3. App reregisters with actual HashId
+-- 4. HMI responds with erroneous resultCode for one interior vehicle data type
+-- SDL does:
+-- 1. process unsuccess response from HMI
+-- 2. remove already restored data
+-- 3. respond RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Resumption/Handling_errors_from_HMI/commonResumptionErrorHandling')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+-- [[ Local Function ]]
+local function getValue(pModuleType, pSubscribe)
+  local moduleIdValue = common.getModuleControlData(pModuleType, 1).moduleId
+  return { moduleType = pModuleType, moduleId = moduleIdValue, subscribe = pSubscribe }
+end
+
+local function getExpectedData()
+  local expectedData = {}
+  for _, moduleTypeValue in pairs(common.rcModuleTypes) do
+    table.insert(expectedData, getValue(moduleTypeValue, true))
+    table.insert(expectedData, getValue(moduleTypeValue, false))
+  end
+  return expectedData
+end
+
+local function checkResumptionData()
+  local expectedData = getExpectedData()
+  local expectedNumber = 2*#common.rcModuleTypes-1 --`-1` because SDL does not send revert request for data
+                                                   -- with erroneous response
+  common.getHMIConnection():ExpectRequest("RC.GetInteriorVehicleData")
+  :Do(function(exp, data)
+      common.log(data.method .. ", moduleType: " .. data.params.moduleType)
+      if exp.occurences == #common.rcModuleTypes then
+        for k, value in pairs(expectedData) do
+          if common.isTableEqual(value, getValue(data.params.moduleType, false)) then
+            table.remove(expectedData, k)
+          end
+        end
+        local function sendResponse()
+          common.log(data.method .. ": GENERIC_ERROR, moduleType: " .. data.params.moduleType)
+          common.getHMIConnection():SendError(data.id, data.method, "GENERIC_ERROR", "info message")
+        end
+        RUN_AFTER(sendResponse, 300)
```

Comment:
@GetmanetsIrina I would propose to use `m.run.runAfter()` from actions module instead `RUN_AFTER`